services/src/main/java/org/keycloak/authentication/authenticators/access/DenyAccessAuthenticator.java

```diff
@@ -46,9 +46,12 @@ public void authenticate(AuthenticationFlowContext context) {
                 .orElse(Messages.ACCESS_DENIED);
 
         context.getEvent().error(Errors.ACCESS_DENIED);
-        Response challenge = context.form()
```

Comment:
Isn't possible to keep the current construct/API and be smarter about how the response should be created based on the media type when actually calling `createErrorPage`?

Reply:
Hi @pedroigor 
Sounds a good idea.
Let me try it.